src/HCSR04.cpp

```diff
@@ -27,10 +28,14 @@ double UltraSonicDistanceSensor::measureDistanceCm(float temperature) {
     digitalWrite(triggerPin, HIGH);
     delayMicroseconds(10);
     digitalWrite(triggerPin, LOW);
+    double speedOfSoundInCmPerMs = 0.03313 + 0.0000606 * temperature; // Cair ≈ (331.3 + 0.606 ⋅ ϑ) m/s
+
+    // Compute max delay based on max distance with 25% margin in microseconds
+    uint32_t timeout = 2000 * speedOfSoundInCmPerMs * maxDistanceCm * 25 / 100;
```

Comment:
I would name the variable `maxDistanceDurationMicroSec` and then I would use it below as third argument like this `maxDistanceDurationMicroSec * 1.25`.
Also, shouldn't it be:
`maxDistanceDurationMicroSec = 2 * maxDistanceCm / speedOfSoundInCmPerMs`?

Reply:
You're right, I forgot speedOfSoundInCmPerMs is already a double variable so no need to keep integer calulations but I think it's a bad idea to pass a floating point value to pulseIn.
In an ideal world I think maxDistanceDurationMicroSec need to be rounded and casted to ensure portability.

If I made no mistake speedOfSound unit is Cm per milliseconds, you have to multiply by 1000 to pass microseconds timeout parameter to [pulseIn](https://www.arduino.cc/reference/en/language/functions/advanced-io/pulsein/).

Saying that I have a doubt : 331.3 m/s = 0.3313 m/ms = 0.00033313 m/µs
Isn't there a mistake in caclulation of speedOfSoundInCmPerMs or I have missed something ?

I'll add a commit to rename variable as you wish.

Reply:
Take a look at line 39: you will see `double distanceCm = durationMicroSec / 2.0 * speedOfSoundInCmPerMs;`. Assuming this works correctly, it means that `Ms` stands for microseconds! But it is very confusing I agree -> if we are dealing with microseconds, we should be explicit about it and certainly not use `ms` as abbreviation for it, since `ms` is normally used for milliseconds.
So, let's also rename `speedOfSoundInCmPerMs` to `speedOfSoundInCmPerMicroSec`, and let's not use `ms` anywhere but let's instead use `MicroSec`.

Regarding rounding -> makes sense! I am not sure what pulseIn takes, I will leave that to you, but I don't have anything against rounding `maxDistanceDurationMicroSec`, I don't think 1 microsecond will change much.

Reply:
You're definitely right, I forgot m to cm conversion.

Renaming ok and max distance calulation fixed (inverted units).

I made some tests with sensors and everything is working.